### ISSUE TITLE
WinForm System.StackOverflowException with .Net Framework 4.6.1

### ISSUE THREAD

**Issue by poulinjf** [microsoft/appcenter-sdk-dotnet#1326]
**Description**

We have a WinForm project that we're trying to integrate AppCenter on top of, but we're getting an System.StackOverflowException as soon as we launch the app.

Looks like it's trying to locate a file named Common.cs that doesn't exist in the execution directory:
Locating source for 'XXX\obj\Debug\Common.cs'. (No checksum.)
The file 'XXX\obj\Debug\Common.cs' does not exist.
Looking in script documents for 'XXX\obj\Debug\Common.cs'...
Looking in the projects for 'XXX\obj\Debug\Common.cs'.
The file was not found in a project.
The debug source files settings for the active solution indicate that the debugger will not ask the user to find the file: XXX\obj\Debug\Common.cs.
The debugger could not locate the source file 'XXX\obj\Debug\Common.cs'.

**Repro Steps**
Create a WinForm project with Visual Studio 2019 using .Net Framework 4.6.1
Install Microsoft.AppCenter.Analytics and Microsoft.AppCenter.Craches from Nuget

Add required librairies to your program.cs file
using Microsoft.AppCenter;
using Microsoft.AppCenter.Analytics;
using Microsoft.AppCenter.Crashes;

Add the AppCenter call before you actually start the app
AppCenter.Start("XXX", typeof(Analytics), typeof(Crashes));

Start Debug

**Details**
**What is your app platform (Xamarin.Android or Xamarin.iOS or UWP)?**
WinForm
**Which SDK version are you using?**
2.6.4
**Which OS version did you experience the issue on?**
Windows 10
**What third party libraries are you using?**
 ```
 <package id="Autoupdater.NET.Official" version="1.5.8" targetFramework="net461" />
  <package id="CommandLineParser" version="2.7.82" targetFramework="net461" />
  <package id="Costura.Fody" version="4.1.0" targetFramework="net461" />
  <package id="DotNetZip" version="1.13.5" targetFramework="net461" />
  <package id="Flurl" version="2.8.2" targetFramework="net461" />
  <package id="Fody" version="6.0.6" targetFramework="net461" developmentDependency="true" />
  <package id="Microsoft.Azure.KeyVault.Core" version="3.0.4" targetFramework="net461" />
  <package id="Microsoft.Bcl" version="1.1.10" targetFramework="net461" />
  <package id="Microsoft.Bcl.Async" version="1.0.168" targetFramework="net461" />
  <package id="Microsoft.Bcl.Build" version="1.0.21" targetFramework="net461" />
  <package id="Microsoft.Rest.ClientRuntime" version="2.3.21" targetFramework="net461" />
  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.19" targetFramework="net461" />
  <package id="Newtonsoft.Json" version="12.0.3" targetFramework="net461" />
  <package id="ObjectListView.Official" version="2.9.1" targetFramework="net461" />
  <package id="Selenium.Support" version="3.141.0" targetFramework="net461" />
  <package id="Selenium.WebDriver" version="3.141.0" targetFramework="net461" />
  <package id="Sentry" version="1.2.0" targetFramework="net461" />
  <package id="Sentry.PlatformAbstractions" version="1.0.0" targetFramework="net461" />
  <package id="Sentry.Protocol" version="1.0.6" targetFramework="net461" />
  <package id="System.Buffers" version="4.4.0" targetFramework="net461" />
  <package id="System.Collections.Immutable" version="1.7.0" targetFramework="net461" />
  <package id="System.Memory" version="4.5.3" targetFramework="net461" />
  <package id="System.Numerics.Vectors" version="4.4.0" targetFramework="net461" />
  <package id="System.Runtime.CompilerServices.Unsafe" version="4.5.2" targetFramework="net461" />
  <package id="System.Runtime.InteropServices.RuntimeInformation" version="4.3.0" targetFramework="net461" />
  <package id="WindowsAzure.Storage" version="9.3.3" targetFramework="net461" />
```

**Comment by russelarms** [comment]
Hi there, thanks for reaching out!
We are looking into this.

**Comment by olmisa** [comment]
Hi @poulinjf, I was not able to reproduce this issue with provided repro steps. Also, I don't see a file with the name 'Common.cs' in 'App Center' repository. 
I have created a sample project (https://github.com/olmisa/TestWinFromsAppCenterCrash). Could you, please, try to reproduce this issue on this project?

**Comment by poulinjf** [comment]
Hi Olmisa,
thanks for getting back to me. I'm able to use your project without any issue, and it looks like it's related to Fody, as I'm able to successfully compile without Costura Fody installed, but Costura Fody is working perfectly fine without AppCenter installed.

Enabling verbose logs without Fody installed gets me this
2020-02-14 16:09:14.684 [AppCenter] DEBUG: Invoke FilteringLog event for channel 'core'
2020-02-14 16:09:14.716 [AppCenter] ERROR: Error persisting log
Microsoft.AppCenter.Storage.StorageException: The storage operation failed ---> System.TypeInitializationException: The type initializer for 'SQLite.SQLiteConnection' threw an exception. ---> System.IO.FileLoadException: Could not load file or assembly 'SQLitePCLRaw.batteries_v2, Version=2.0.2.669, Culture=neutral, PublicKeyToken=8226ea5df37bcae9' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040) ---> System.IO.FileLoadException: Could not load file or assembly 'SQLitePCLRaw.batteries_v2, Version=1.0.0.0, Culture=neutral, PublicKeyToken=8226ea5df37bcae9' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   --- End of inner exception stack trace ---
   at SQLite.SQLiteConnection..cctor()
   --- End of inner exception stack trace ---
   at SQLite.SQLiteConnection..ctor(String databasePath, SQLiteOpenFlags openFlags, Boolean storeDateTimeAsTicks)
   at SQLite.SQLiteConnectionPool.Entry..ctor(SQLiteConnectionString connectionString, SQLiteOpenFlags openFlags)
   at SQLite.SQLiteConnectionPool.GetConnection(SQLiteConnectionString connectionString, SQLiteOpenFlags openFlags)
   at SQLite.SQLiteAsyncConnection.<InsertAsync>c__AnonStorey1.<>m__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AppCenter.Storage.Storage.<>c__DisplayClass11_0.<PutLog>b__0()
   at Microsoft.AppCenter.Storage.Storage.<>c__DisplayClass24_0.<AddTaskToQueue>b__0()
   --- End of inner exception stack trace ---
   at Microsoft.AppCenter.Storage.Storage.<>c__DisplayClass24_0.<AddTaskToQueue>b__0()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AppCenter.Channel.Channel.<PersistLogAsync>d__40.MoveNext()

**Comment by MatkovIvan** [comment]
Hi @poulinjf, 

This is `SQLite` version conflict, the current version uses sqlite 1.x that incompatible with 2.x. Please downgrade version of `SQLite` library to make it work. The next version of SDK will use 2.x, the release is planned this month.
We already have opened issue for it, so I'm closing this as a duplicate.
Related issues: #1313 #1085 microsoft/appcenter#1460

**Comment by poulinjf** [comment]
For those whom are dealing with the same issue, don't let Nuget install those dependencies packages for you, but install them manually instead.

First, download those old SQLite packages
`  <package id="sqlite-net-pcl" version="1.3.3" targetFramework="net461" />
  <package id="SQLitePCLRaw.bundle_green" version="1.1.14" targetFramework="net461" />
  <package id="SQLitePCLRaw.core" version="1.1.14" targetFramework="net461" />
  <package id="SQLitePCLRaw.lib.e_sqlite3.linux" version="1.1.14" targetFramework="net461" />
  <package id="SQLitePCLRaw.lib.e_sqlite3.osx" version="1.1.14" targetFramework="net461" />
  <package id="SQLitePCLRaw.lib.e_sqlite3.v110_xp" version="1.1.14" targetFramework="net461" />
  <package id="SQLitePCLRaw.lib.e_sqlite3.v140" version="1.1.14" targetFramework="net461" />
  <package id="SQLitePCLRaw.provider.e_sqlite3.net45" version="1.1.14" targetFramework="net461" />`

Then, install Microsoft AppCenter packages
`  <package id="Microsoft.AppCenter" version="2.6.2" targetFramework="net461" />
  <package id="Microsoft.AppCenter.Analytics" version="2.6.2" targetFramework="net461" />
  <package id="Microsoft.AppCenter.Crashes" version="2.6.2" targetFramework="net461" />`

**Comment by guperrot** [comment]
We just released version 3.0.0 of AppCenter with new updated sqlite dependencies (to sqlite 2.x).
